deploy/deploy-centralvenv.sh

```diff
@@ -0,0 +1,1000 @@
+wmCfgRepo="https://:@gitlab.cern.ch:8443/cmsweb-k8s/services_config.git"  # WMCore config Repo
+wmCfgBranch="test"                                                        # WMCore config branch
+wmTag=""                                                                  # wmcore tag Default: no tag
+serPatch=""                                                               # a list of service patches to be applied
+runFromSource=false                                                       # a bool flag indicating run from source
+vmName=""                                                                 # hostname for central services
+vmName=${vmName%%.*}
+pipIndex="prod"                                                           # pypi Index to use
+verboseMode=false
+assumeYes=false
+noVenvCleanup=false
+secString=""
+
+# NOTE: We are about to stick to Python3 solely from now on. So if the default
+#       python executable for the system we are working on (outside the virtual
+#       environment) is linked to Python2, then we should try creating the environment
+#       with `python3' instead of `python`. If this link is not present, we simply
+#       fail during virtual environment creation. Once we are inside the virtual
+#       environment the default link should always point to e Python3 executable
+#       so the `pythonCmd' variable shouldn't be needed any more.
+
+pythonCmd=python
+[[ $(python -V 2>&1) =~ Python\ *2.* ]] && pythonCmd=python3
+
+### Searching for the mandatory and optional arguments:
+# export OPTIND=1
+while getopts ":t:c:r:b:g:j:d:p:m:l:i:snvyh" opt; do
+    case ${opt} in
+        d)
+            venvPath=$OPTARG
+            venvPath=$(_realPath $venvPath) ;;
+        t)
+            wmTag=$OPTARG ;;
+        c)
+            vmName=$OPTARG
+            vmName=${vmName%%.*} ;;
+        r)
+            wmSrcRepo=$OPTARG ;;
+        b)
+            wmSrcBranch=$OPTARG ;;
+        g)
+            wmCfgRepo=$OPTARG ;;
+        j)
+            wmCfgBranch=$OPTARG ;;
+        p)
+            serPatch=$OPTARG ;;
+        m)
+            secString=$OPTARG ;;
+        l)
+            serviceList=$OPTARG ;;
+        i)
+            pipIndex=$OPTARG ;;
+        s)
+            runFromSource=true ;;
+        n)
+            noVenvCleanup=true ;;
+        v)
+            verboseMode=true ;;
+        y)
+            assumeYes=true ;;
+        h)
+            help
+            exit 0 ;;
+        \? )
+            msg="Invalid Option: -$OPTARG"
+            usage "$msg" ;;
+        : )
+            msg="Invalid Option: -$OPTARG requires an argument"
+            usage "$msg" ;;
+    esac
+done
+
+$verboseMode && set -x
+
+# Swap noVenvCleanup flag with venvCleanup to avoid double negation and confusion:
+venvCleanup=true && $noVenvCleanup && venvCleanup=false
+
+# check for mandatory parameters:
+[[ -z $vmName ]] && usage "Missing mandatory argument: -c <central_services>"
```

Comment:
please re-think this part and add ability to avoid mandatory central services. Instead, provide user configs.

Reply:
removed